### PR TITLE
Support for reactive Aerospike repository

### PR DESCRIPTION
**Pull request by igor-ermolenko**
Created stub repository and autoconfiguration.
All repository methods implementations and tests will be added in the following pool requests.

### CODE REVIEW

**Comment by igor-ermolenko** [comment]
@Aloren @BrianNichols could you please review this pull request? Thank you!

**Comment by arrowplum** [comment]
Since we are adding @Aloren as a maintainer here please request her as reviewer and to make releases rather than Brian Nichols.

I have added her already to reviewer list (and given maintainer permissions.)

**Comment by arrowplum** [comment]
BTW @Aloren should be able to merge it in but we are still figuring out how to give the permissions for pushing up to maven.

File: src/main/java/org/springframework/data/aerospike/repository/support/SimpleReactiveAerospikeRepository.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+    @Override
+    public <S extends T> Mono<S> save(S entity) {
+        Assert.notNull(entity, "Cannot save NULL entity");
+        return operations.save(entity);
+    }
+
+    @Override
+    public <S extends T> Flux<S> saveAll(Iterable<S> entities) {
+        return Flux.empty();
```
Comment (tdanylchuk) [inline src/main/java/org/springframework/data/aerospike/repository/support/SimpleReactiveAerospikeRepository.java]: throw unsupported exceptions

File: src/main/java/org/springframework/data/aerospike/repository/config/EnableReactiveAerospikeRepositories.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2015 the original author or authors.
```
[the remark this answers is not in the document]
Comment (igor-ermolenko) [reply src/main/java/org/springframework/data/aerospike/repository/config/EnableReactiveAerospikeRepositories.java]: Done

File: src/main/java/org/springframework/data/aerospike/repository/support/SimpleReactiveAerospikeRepository.java (hunk 1 above)
Comment (igor-ermolenko) [reply src/main/java/org/springframework/data/aerospike/repository/support/SimpleReactiveAerospikeRepository.java]: Done

File: src/main/java/org/springframework/data/aerospike/repository/query/ReactiveAerospikePartTreeQuery.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	
+	/* (non-Javadoc)
+	 * @see org.springframework.data.repository.query.RepositoryQuery#execute(java.lang.Object[])
+	 */
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	@Override
+	public Object execute(Object[] parameters) {
+		ParametersParameterAccessor accessor = new ParametersParameterAccessor(queryMethod.getParameters(), parameters);
+		Query query = prepareQuery(parameters, accessor);
+		return aerospikeOperations.find(query, queryMethod.getEntityInformation().getJavaType());
```
Comment (Aloren) [inline src/main/java/org/springframework/data/aerospike/repository/query/ReactiveAerospikePartTreeQuery.java]: is this blocking operation? i do not think this is a good idea to have blocking operations exposed under reactive interface.

File: src/test/java/org/springframework/data/aerospike/repository/support/SimpleReactiveAerospikeRepositoryTest.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	Person testPerson;
+	List<Person> testPersons;
+
+	@Rule
+	public ExpectedException exception = ExpectedException.none();
+
+
+	@Before
+	public void setUp() throws Exception {
+//		when(metadata.getJavaType()).thenReturn(Person.class);
```
Comment (Aloren) [inline src/test/java/org/springframework/data/aerospike/repository/support/SimpleReactiveAerospikeRepositoryTest.java]: please remove commented code

Comment (igor-ermolenko) [reply src/test/java/org/springframework/data/aerospike/repository/support/SimpleReactiveAerospikeRepositoryTest.java]: Commented and unused stuff removed

File: src/main/java/org/springframework/data/aerospike/repository/query/ReactiveAerospikePartTreeQuery.java (hunk 3 above)
Comment (igor-ermolenko) [reply src/main/java/org/springframework/data/aerospike/repository/query/ReactiveAerospikePartTreeQuery.java]: No, it's non blocking: find() returns Flux.
<T> Flux<T> find(Query query, Class<T> type)

**Comment by arrowplum** [comment]
Hi @Aloren if everything is looking good to you, you should merge the changes in. Still working on getting you access to push to maven

**Comment by Aloren** [comment]
Thanks, @arrowplum! We were discussing with @igor-ermolenko that these changes should not be merged to master directly as they are not complete yet, we are going to use separate branch for the feature right now and once the feature is done we are going to merge everything to master.

**Pull request merged**